homeassistant/components/alert/entity.py

```diff
@@ -195,9 +195,12 @@ async def async_turn_on(self, **kwargs: Any) -> None:
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Async Acknowledge alert."""
-        LOGGER.debug("Acknowledged Alert: %s", self._attr_name)
-        self._ack = True
-        self.async_write_ha_state()
+        if self._can_ack:
+            LOGGER.debug("Acknowledged Alert: %s", self._attr_name)
+            self._ack = True
+            self.async_write_ha_state()
+        else:
+            LOGGER.debug("Alert Acknowledgement Blocked: %s", self._attr_name)
```

Comment:
```suggestion
            LOGGER.info("Alert Acknowledgement Blocked: %s", self._attr_name)
```
I think this should go in the log so that people can figure out why it's not working. They might have forgotten that they set `can_acknowledge` to false.

Reply:
Good point. Probably actually a warning though?

Reply:
Resolving in favor of thread with Frenck https://github.com/home-assistant/core/pull/139483#discussion_r1975758973

Reply:
Resolving in favor of thread with Frenck https://github.com/home-assistant/core/pull/139483#discussion_r1975758973

Edit: my browser was cached, looks like he already resolved it.